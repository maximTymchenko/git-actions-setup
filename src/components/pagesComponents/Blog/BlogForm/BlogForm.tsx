import React, { FC, useState } from 'react';
import './BlogForm.scss';
import classNames from 'classnames';
import Button from '../../../common/Button';
import arrowRight from '../../../../assets/img/arr-right.svg';
import * as EmailValidator from 'email-validator';
import validator from 'validator';

const BlogForm: FC = () => {
    const [emailClass, setEmailClass] = useState<boolean>(false);

    const [emailError, setEmailError] = useState('');
    const validateEmail = (e) => {
        var email = e.target.value;

        if (validator.isEmail(email)) {
            setEmailError('Valid Email :)');
            setEmailClass(false);
        } else {
            setEmailError('Enter valid Email!');
            setEmailClass(true);
        }
    };

    const removeReloadPageWithClick = (e) => {
        e.preventDefault();
    };
    return (
        <form onSubmit={removeReloadPageWithClick} className="blog-header__form">
            <div className="blog-header__input-wrapper">
                <input
                    onChange={(e) => validateEmail(e)}
                    className={classNames('blog-header__input', {
                        'blog-header__input--error': emailClass,
                    })}
                    type="text"
                    name="email"
                    placeholder="Your email"
                />
                <span
                    className={classNames('blog-header__error', {
                        'blog-header__error--show': emailClass,
                    })}
                >
                    Check email
                </span>
            </div>
            <div className="blog-header__button">
                <Button icon={arrowRight} type="secondary">
                    Subscribe
                </Button>
            </div>
            {/* {emailError} */}
        </form>
    );
};

export default BlogForm;
