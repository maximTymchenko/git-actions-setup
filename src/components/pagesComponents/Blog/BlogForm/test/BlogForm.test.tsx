import React from 'react';
import BlogForm from '../BlogForm';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<BlogForm />);
    expect(getByTestId('blog-form')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<BlogForm />).toJSON();
    expect(tree).toMatchSnapshot();
});
