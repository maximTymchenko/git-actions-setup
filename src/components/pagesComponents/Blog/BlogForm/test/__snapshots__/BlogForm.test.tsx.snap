// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<form
  className="blog-header__form"
  data-testid="blog-form"
  onSubmit={[Function]}
>
  <div
    className="blog-header__input-wrapper"
  >
    <input
      className="blog-header__input"
      name="email"
      placeholder="Your email"
      type="text"
    />
    <span
      className="blog-header__error"
    >
      Check email
    </span>
  </div>
  <div
    className="blog-header__button"
  >
    <button
      className="button button--secondary button--undefined button--undefined"
      data-testid="button"
    >
      Subscribe
      <img
        className="button__icon button__icon--undefined"
        data-testid="button__icon"
        src="test-file-stub"
      />
    </button>
  </div>
</form>
`;
